app/models/service/resource_linking.rb

```diff
@@ -0,0 +1,37 @@
+module Service::ResourceLinking
+  extend ActiveSupport::Concern
+
+  def add_provider_vms(options)
+    # required keys in options hash:
+    #   :target_id, the id of the provider to refresh
+    #   :uid_ems_array, an array of VMs' uid_ems
+    #
+    options = {
+      :target_class => "ExtManagementSystem",
+      :name         => "Linking VMs to service #{name} ID: [#{id}]",
+      :userid       => evm_owner.userid,
+      :sync_key     => guid,
+      :service_id   => id,
+      :zone         => my_zone
```

Comment:
Is `zone` needed for the provider refresh?  If so this should be the providers zone not the service.  Otherwise it would be good to eliminate all together.